zellij-server/src/tab.rs

```diff
@@ -318,26 +322,50 @@ impl Tab {
         }
         let mut new_pids = new_pids.iter();
         for (layout, position_and_size) in positions_and_size {
-            // A plugin pane
-            if let Some(plugin) = &layout.plugin {
-                let (pid_tx, pid_rx) = channel();
-                self.senders
-                    .send_to_plugin(PluginInstruction::Load(pid_tx, plugin.clone()))
-                    .unwrap();
-                let pid = pid_rx.recv().unwrap();
-                let new_plugin = PluginPane::new(
-                    pid,
-                    *position_and_size,
-                    self.senders.to_plugin.as_ref().unwrap().clone(),
-                );
-                self.panes.insert(PaneId::Plugin(pid), Box::new(new_plugin));
-                // Send an initial mode update to the newly loaded plugin only!
-                self.senders
-                    .send_to_plugin(PluginInstruction::Update(
-                        Some(pid),
-                        Event::ModeUpdate(self.mode_info.clone()),
-                    ))
-                    .unwrap();
+            if let Some(runnable) = &layout.run {
+                match runnable {
+                    // A plugin pane
+                    Run::Plugin(Some(plugin)) => {
+                        let (pid_tx, pid_rx) = channel();
+                        self.senders
+                            .send_to_plugin(PluginInstruction::Load(pid_tx, plugin.clone()))
+                            .unwrap();
+                        let pid = pid_rx.recv().unwrap();
+                        let new_plugin = PluginPane::new(
+                            pid,
+                            *position_and_size,
+                            self.senders.to_plugin.as_ref().unwrap().clone(),
+                        );
+                        self.panes.insert(PaneId::Plugin(pid), Box::new(new_plugin));
+                        // Send an initial mode update to the newly loaded plugin only!
+                        self.senders
+                            .send_to_plugin(PluginInstruction::Update(
+                                Some(pid),
+                                Event::ModeUpdate(self.mode_info.clone()),
+                            ))
+                            .unwrap();
+                    }
+                    // A command pane
+                    Run::Command(command) => {
+                        let cmd = TerminalAction::RunCommand(command.clone());
+                        self.senders
+                            .send_to_pty(PtyInstruction::SpawnTerminal(Some(cmd)))
+                            .unwrap();
+
+                        // I would expect something like this to work,
+                        // but I must still be overlooking something.
+                        //let (_, pid) = self.os_api.spawn_terminal(Some(cmd));
```

Comment:
This was my most recent attempt.
As far as I understand we want to:
- create the terminal -> get the pid
- use the pid to create a new terminal struct and pass the correct size
- then insert the terminal into the tab struct